swap_meet/vendor.py

```diff
@@ -0,0 +1,97 @@
+from swap_meet.item import Item
+
+class Vendor:
+
+    def __init__(self, inventory = None):
+        if inventory is None:
+            self.inventory = []
+        else:
+            self.inventory = inventory
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+    
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        return False
+    
+    def get_by_category(self, category_name):
+        items = []
+        for item in self.inventory:
+            if category_name == item.category:
+                items.append(item)
+        return items
+    
+    def swap_items(self, vendor_friend, my_item, their_item):  
+        if my_item not in self.inventory or their_item not in vendor_friend.inventory:
+            return False
+        else:
+            self.remove(my_item)
+            vendor_friend.remove(their_item)
+            vendor_friend.add(my_item)
+            self.add(their_item)
+            return True
+    
+    def swap_first_item(self, vendor_friend):
+        if len(self.inventory) == 0 or len(vendor_friend.inventory) == 0:
+            return False
+        else:
+            item_a = self.inventory[0]
+            item_d = vendor_friend.inventory[0]
+            result = self.swap_items(vendor_friend, item_a, item_d)
```

Comment:
we could get rid of these two variables if we wanted!

```suggestion
            result = self.swap_items(vendor_friend, self.inventory[0], vendor_friend.inventory[0])
```